nunit/MVP-Advance/SeleniumNUnit/Pages/ManageListings.cs

```diff
@@ -36,9 +42,93 @@ internal class ManageListings
         private IWebElement btnSave => driver.FindElement(By.XPath("//input[@value='Save']"));
         #endregion
 
-        internal void AddListing()
+        ShareSkill shareSkillObj;
+
+        //Add a skill
+        internal void AddListing(int rowNumber, string worksheet)
+        {
+            shareSkillObj = new ShareSkill();
+            btnShareSkill.Click();
+            wait(2);
+            shareSkillObj.EnterShareSkill(rowNumber, worksheet);
+            wait(3);
+        }
+
+        //Verify add & edit
+        internal void ViewListing(int rowNumber, string worksheet)
+        {
+
+            //Click on ManageListing
+            GoToManageListings();
+            WaitForElement(driver, By.XPath(eTable), 5);
+
+            //Read data
+            ExcelLib.PopulateInCollection(Base.ExcelPath, worksheet);
+            string expectedTitle = ExcelLib.ReadData(rowNumber, "Title");
+
+            //Click on button View
+            string e_View = "//div[@id='listing-management-section']//tbody/tr[" + GetTitleIndex(expectedTitle) + "]/td[8]/div/button[1]";
+            IWebElement btnView = driver.FindElement(By.XPath(e_View));
+            btnView.Click();
+
+            wait(2);
+        }
+
+        internal void EnterShareSkill_Invalid(int testData, string worksheet)
         {
+            shareSkillObj = new ShareSkill();
+            //Click on button ShareSkill
+            btnShareSkill.Click();
+            wait(1);
+
+            //Enter invalid data
+            shareSkillObj.EnterShareSkill_InvalidData(testData, "NegativeTC");
+            Thread.Sleep(2000);
+        }
 
+        //Functions to check title is existing and return title's position in manage listing
+        internal string GetTitleIndex(string expectedTitle)
+        {
+            //Check if there is no listing's title
+            string recordIndex = "";
+            int titleCount = Titles.Count();
+            if (titleCount.Equals(0))
+            {
+                Assert.Ignore("There is no listing record.");
+            }
+            else
+            {
+                //Find title: Break loop when finding a title. Output: recordIndex
+                for (int i = 0; i < titleCount; i++)
+                {
+                    string actualTitle = Titles[i].Text;
+                    if (actualTitle.Equals(expectedTitle))
+                    {
+                        recordIndex = (i + 1).ToString();
+                        break;
+                    }
+                }
+                //If title-to-delete is not found
+                if (recordIndex.Equals(""))
+                {
+                    Assert.Ignore("Listing '" + expectedTitle + "' is not found.");
```

Comment:
Can you please move the asserts into the tests, if the are testing multiple things then I suggest you return a state for example you asserting ignore instead return null and then in the test check if its null then assert ignore. This way i do not need to look at page logic